### DISCUSSION TITLE
LXQt configs disappeared and task manager does not display window information

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2288] [Q&A]
I was trying out an alpha build of OpenMandriva Rolling with LXQt. Two bug reports were filed at OpenMandrivaAssociation on GitHub. 

One was about specific configs disappearing, which I did not experience.  https://github.com/OpenMandrivaAssociation/distribution/issues/2815

The other is where the Task Manager isn't displaying any window information. If a window is minimized, you can't recover it. This is an issue I experienced. https://github.com/OpenMandrivaAssociation/distribution/issues/2816

Could someone look at the two bug reports and possibly provide information on what might be happening? Both bug reports have screenshots. The individual who posted the bug reports does not use LXQt.

Thank you.

**Comment by stefonarch** [comment]
For the second one: can you post the [taskbar] section from `~/.config/lxqt/panel.conf` file?

**Reply by edwardp3** [reply]
```
[taskbar]
alignment=Right
autoRotate=True
buttonHeight=100
buttonStyle=IconText
buttonWidth=400
closeOnMiddleClick=True
groupingEnabled=True
raiseOnCurrentDesktop=false
showDesktopNum=0
showGroupOnHover=true
showOnlyCurrentScreenTasks=false
showOnlyMinimizedTasks=true
showOnlyOneDesktopTasks=false
type=taskbar
```

**Reply by tsujan** [reply]
Self-explanatory → `showOnlyMinimizedTasks=true`

**Reply by edwardp3** [reply]
I changed that entry to `false` and the panel remains empty after a logout/login and a reboot:

![om](https://user-images.githubusercontent.com/73129450/181973554-2e0ebadd-39c2-4759-8e24-b539ebd06aae.png)

**Reply by tsujan** [reply]
> I changed that entry to false

Panel will overwrite it (unless you stop it from Session Settings during the manual change). You should change it in the panel config. No reboot or logout is needed.

**Reply by edwardp3** [reply]
I changed it in the config file, because none of the settings in the Config Panel (attached) matched or came close to the name the setting in the config file, so I did not know what to change.

![om-lxqt-panel-config](https://user-images.githubusercontent.com/73129450/181995329-3a421821-ad26-4d22-b2e0-3239c850bc9f.png)

@stefonarch  I went through all of the Themes and except for the theme change itself, the panel remained unchanged, completely empty, not even any borders around what would have been a window title. As for invisible text, I do not even know where to begin to look for that.

**Comment by stefonarch** [comment]
> If a window is minimized, you can't recover it.

 You should with this setting.

Anyway better change settings in the GUI not in the file.
Not an color issue? No icons available and invisible text?

**Comment by tsujan** [comment]
BTW, if your distro ships a settings file with that option turned on, you might want to ask them to change it.

**Reply by edwardp3** [reply]
Changing that setting obviously didn't work.

This is the Task Manager Settings screen. I also tried everything here, without success.

![om-task-manager](https://user-images.githubusercontent.com/73129450/181995691-b9fcadd5-1fc0-4d91-994b-c360f1cdab30.png)

**Comment by stefonarch** [comment]
I hardly can immagine how this can happen... window manager is openbox I assume from the screen.
Try xfwm4 maybe?

**Reply by edwardp3** [reply]
I installed the image in another virtual machine, made the same change as above to the panel.conf file. Upgraded the repositories to Cooker, which gave me LXQt 1.1.0. Openbox was the window manager. Installed xfwm4, no change in behavior of the panel/task manager. 

Unless there are other ideas???

**Reply by tsujan** [reply]
> Unless there are other ideas?

Not an idea but a fact: the config part you attached contradicts your last screenshot. It may have various reasons, like wrong file permissions, distro patches, bad build,...

**Reply by stefonarch** [reply]
At this point I think bad compilation/packages. Still a mystery though.

**Reply by edwardp3** [reply]
> At this point I think bad compilation/packages. Still a mystery though.

I'm inclined to agree. OM Rolling is something new for them. I asked the individual who posted the bug reports to review this thread.

**Comment by benbullard79** [comment]
I installed `xfwm4` in the OMLx 'Live' iso and changed the window manager in ~/.config/session.conf and /usr/share/lxqt/session.conf. Then logged out, restarted X and logged in and still do not see the task manager or desktop switcher.

![Screenshot_20230107_123018](https://user-images.githubusercontent.com/5704386/211165603-5654c98b-8f69-44f2-be40-12a49299858b.jpg)

I am not sure if this is all there is to testing a different window manager for a LXQt system set up to use openbox. (I am not a developer and don't know code well at all.)

**Comment by benbullard79** [comment]
Is this possibly caused by this: https://github.com/lxqt/lxqt-session/issues/427 ?

**Reply by tsujan** [reply]
https://github.com/OpenMandrivaAssociation/distribution/issues/2816#issuecomment-1208157046 says that, for whatever reason, your distro has chosen to set`showOnlyMinimizedTasks=true` in `panel.conf` (LXQt sets it to `false`). If so, you shouldn't see any task button unless you minimize a window.

**Comment by benbullard79** [comment]
That was changed in this patch: https://github.com/OpenMandrivaAssociation/lxqt-panel/blob/master/lxqt-panel-0.8.0-omv-settings.patch and the problem persists.

This is ~/.config/lxqt/panel.conf from most recent LXQt iso:

[panel.conf.txt](https://github.com/lxqt/lxqt/files/10367161/panel.conf.txt)

If there is anything wrong there let us know.

**Reply by tsujan** [reply]
> That was changed in this patch...

Then something is definitely wrong. It's unlikely that its cause is in LXQt because we don't see it in Arch, Manjaro, Debian,... It may be caused by bad build(s) or incompatible libraries; sorry, I can't tell for sure.

BTW, your attached conf files seem OK.

**Reply by benbullard79** [reply]
Thanks, good to know the conf files are OK. FWIW my thinking on this has been that it was something with OM not LXQT, just don't know what yet. Will keep looking.

Also the configs OP was referring to have been fixed. The only issue I am aware of now is task manager and desktop switcher are not visible.

**Reply by tsujan** [reply]
>  The only issue I am aware of now is task manager and desktop switcher are not visible.

For now, the only thing that comes to my mind is that there may be a problem in your installed `KWindowSystem`. Task Manager and Desktop Switcher both need it to work properly.

**Comment by benbullard79** [comment]
Uh, to be thorough perhaps should also post current /usr/share/lxqt/panel.conf just in case something is amiss there.

[⁄usr⁄share⁄lxqt⁄panel.conf.txt](https://github.com/lxqt/lxqt/files/10367170/usr.share.lxqt.panel.conf.txt)

Edit: The desktop switcher is also invisible, I believe that should always be visible.
